### PR TITLE
Refactor lean-analytics.swig to use REST API to avoid conflicts with valine

### PR DESCRIPTION
**Pull request by LEAFERx**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes have been added (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [x] Docs have been added / updated (for bug fixes / features).

## PR Type
**What kind of change does this PR introduce?**  

- [x] Bugfix.
- [ ] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number(s): #376 

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

File: layout/_third-party/analytics/lean-analytics.swig [hunk 1]
```diff
@@ -49,66 +45,60 @@
...
 
     $(function() {
-      var Counter = AV.Object.extend("Counter");
+      var Counter = function (method, url, data) {
+        return $.ajax({
+          method: method,
+          url: `https://api.leancloud.cn/1.1${url}`,
+          headers: {
+            'X-LC-Id': "{{theme.leancloud_visitors.app_id}}",
+            'X-LC-Key': "{{theme.leancloud_visitors.app_key}}",
```
Comment (Raincal) [inline layout/_third-party/analytics/lean-analytics.swig]: 建议使用 X-LC-Sign 来代替 X-LC-Key
https://leancloud.cn/docs/rest_api.html#hash-1794863242

File: layout/_third-party/analytics/lean-analytics.swig [hunk 2]
```diff
@@ -49,66 +45,60 @@
...
+        console.log('Error:' + error.code + " " + error.message);
       });
     }
 
     $(function() {
-      var Counter = AV.Object.extend("Counter");
+      var Counter = function (method, url, data) {
+        return $.ajax({
+          method: method,
+          url: `https://api.leancloud.cn/1.1${url}`,
```
Comment (Raincal) [inline layout/_third-party/analytics/lean-analytics.swig]: api.leancloud.cn 这个域名在某些地区无法解析，建议换成应用专属的二级域名
https://forum.leancloud.cn/t/leancloud-api/16234

File: layout/_third-party/analytics/lean-analytics.swig (hunk 1 above)
Comment (leeyeh) [reply layout/_third-party/analytics/lean-analytics.swig]: 这一点在保证使用 https 的情况下倒不是必须的。

Comment (LEAFERx) [reply layout/_third-party/analytics/lean-analytics.swig]: md5需要引入额外的库 我觉得在https下安全性是没有问题的 而且其实apikey是暴露的 也没有必要

File: layout/_third-party/analytics/lean-analytics.swig (hunk 2 above)
Comment (LEAFERx) [reply layout/_third-party/analytics/lean-analytics.swig]: 已更换

File: layout/_third-party/analytics/lean-analytics.swig [hunk 3]
```diff
@@ -49,71 +45,68 @@
...
+            .done(function () {
                   var $element = $(document.getElementById(url));
-                  $element.find('.leancloud-visitors-count').text(counter.get('time'));
-                {% endif %}
-              },
-              error: function(counter, error) {
+                  $element.find('.leancloud-visitors-count').text(counter.time + 1);
+            })
+            {% endif %}
+            .fail(function (error) {
```
Comment (leeyeh) [inline layout/_third-party/analytics/lean-analytics.swig]: 这个 error 是 XHR 层面的异常了，LeanCloud 业务层面的异常还需要再拆一层，异常 response body 结构是这样的：

```
{"code":119,"error":"该操作已被禁止，请变更应用选项『数据存储』。"}
```

File: layout/_third-party/analytics/lean-analytics.swig [hunk 4]
```diff
@@ -87,23 +87,26 @@
...
-          },
-          data: data,
-        });
-      };
-      if ($('.leancloud_visitors').length == 1) {
-        addCount(Counter);
-      } else if ($('.post-title-link').length > 1) {
-        showTime(Counter);
-      }
+      $.get('https://app-router.leancloud.cn/2/route?appId=' + "{{theme.leancloud_visitors.app_id}}")
```
Comment (Raincal) [inline layout/_third-party/analytics/lean-analytics.swig]: 其实只要把 `xxx.api.lncld.net` 前缀改成 `APPID.slice(0, 8).toLowerCase()` 就行了，没必要多发一次请求，就是不知道以后会不会变 @leeyeh

Comment (leeyeh) [reply layout/_third-party/analytics/lean-analytics.swig]: 还是有必要的，因为华东节点、美国节点用的域名不是 lncld.net。

Comment (Raincal) [reply layout/_third-party/analytics/lean-analytics.swig]: 了解了~

File: layout/_third-party/analytics/lean-analytics.swig (hunk 3 above)
Comment (LEAFERx) [reply layout/_third-party/analytics/lean-analytics.swig]: 已修改~

**Comment by LEAFERx** [comment]
@Raincal 如果可以请approve或直接merge

**Comment by Raincal** [comment]
@LEAFERx 我没权限啊

**Pull request merged**